nilearn/maskers/tests/test_multi_nifti_labels_masker.py

```diff
@@ -0,0 +1,321 @@
+                        "minimum": np.min(test_values),
+                        "maximum": np.max(test_values),
+                        "standard_deviation": np.std(test_values),
+                        "variance": np.var(test_values)}
+
+    for strategy, expected_result in expected_results.items():
+        masker = MultiNiftiLabelsMasker(labels, strategy=strategy)
+        # Here passing [img, img] within a list because it is multiple subjects
+        # with a 3D object.
+        results = masker.fit_transform([img, img])
+        for result in results:
+            assert result.squeeze() == expected_result
+
+    with pytest.raises(ValueError, match="Invalid strategy 'TESTRAISE'"):
+        MultiNiftiLabelsMasker(
+            labels,
+            strategy="TESTRAISE"
+        )
+
+    default_masker = MultiNiftiLabelsMasker(labels)
+    assert default_masker.strategy == "mean"
+
+
+def test_multi_nifti_labels_masker_resampling():
+    # Test resampling in MultiNiftiLabelsMasker
+    shape1 = (10, 11, 12)
+    affine = np.eye(4)
+
+    # mask
+    shape2 = (16, 17, 18)
+
+    # labels
+    shape3 = (13, 14, 15)
+
+    n_regions = 9
+    length = 3
+
+    # With data of the same affine
+    fmri11_img, _ = data_gen.generate_fake_fmri(shape1, affine=affine,
+                                                length=length)
+    _, mask22_img = data_gen.generate_fake_fmri(shape2, affine=affine,
+                                                length=length)
+
+    labels33_img = data_gen.generate_labeled_regions(shape3, n_regions,
+                                                     affine=affine)
+
+    # Multi-subject example
+    fmri11_img = [fmri11_img, fmri11_img]
+
+    # Test error checking
+    pytest.raises(ValueError, MultiNiftiLabelsMasker, labels33_img,
+                  resampling_target="mask")
+    pytest.raises(ValueError, MultiNiftiLabelsMasker, labels33_img,
+                  resampling_target="invalid")
+
+    # Target: labels
+    masker = MultiNiftiLabelsMasker(labels33_img, mask_img=mask22_img,
+                                    resampling_target="labels")
+
+    masker.fit()
+    np.testing.assert_almost_equal(masker.labels_img_.affine,
+                                   labels33_img.affine)
+    assert masker.labels_img_.shape == labels33_img.shape
+
+    np.testing.assert_almost_equal(masker.mask_img_.affine,
+                                   masker.labels_img_.affine)
+    assert masker.mask_img_.shape == masker.labels_img_.shape[:3]
+
+    transformed = masker.transform(fmri11_img)
+    for t in transformed:
+        assert t.shape == (length, n_regions)
+
+        fmri11_img_r = masker.inverse_transform(t)
+        np.testing.assert_almost_equal(fmri11_img_r.affine,
+                                       masker.labels_img_.affine)
+        assert (fmri11_img_r.shape == (
+                masker.labels_img_.shape[:3] + (length,)))
+
+    # Test with clipped labels: mask does not contain all labels.
```

Comment:
Looking at this really long function, I feel like it could be split in at least 3 different functions. The fact that new data are generated at multiple places is a good indicator. WDYT?